src/main/kotlin/burp/BurpExtender.kt

```diff
@@ -578,13 +579,25 @@ class BurpExtender : IBurpExtender, ITab, ListDataListener, IHttpListener {
 
     private fun loadConfig(): Piper.Config {
         val serialized = callbacks.loadExtensionSetting(EXTENSION_SETTINGS_KEY)
-        return if (serialized == null)
-        {
+        val env = System.getenv(CONFIG_ENV_VAR)
+        try {
+            if (env != null) { 
+                val fmt = if (env.endsWith(".yml") || env.endsWith(".yaml")){
+                    ConfigFormat.YAML 
+                } else {
+                    ConfigFormat.PROTOBUF
+                }
+                val configFile = File(env)
+                return fmt.parse(configFile.readBytes()).updateEnabled(true)
+            } else if (serialized != null) {
+                return Piper.Config.parseFrom(decompress(unpad4(Z85.Z85Decoder(serialized))))
+            } else {
+                throw Exception("Fallback to default config")
```

Comment:
Why the message? Since you catch it with a wildcard `catch` below, nobody will see this. Did you mean to print it to some console?

Reply:
No, it's just meant to be a kind of self-documentation.

Reply:
But if I understand it right, this will then just silently load the default config for any exception (e.g. YAML syntax error). Is this desired/planned behavior?

Reply:
Yes, this is intentional. A lot of things can go wrong during loading files alone, so my thinking was that if *anything* fails, we just go with the defaults. This would be an indication for the user about misconfiguration, but we may need some better signal.